test/engine/IndexScanTest.cpp

```diff
@@ -1010,6 +1010,67 @@ TEST_P(IndexScanWithLazyJoin,
   ASSERT_EQ(joinSideResults.size(), 0);
 }
 
+// _____________________________________________________________________________
+TEST_P(IndexScanWithLazyJoin, prefilterTablesDoesNotSkipOnRepeatingBlock) {
+  std::string kg = "<a> <p> <A> . <b> <p> <B> . <c> <p> <C> . <d> <p> <D> . ";
+  qec_ = getQec(std::move(kg));
+  IndexScan scan = makeScan();
+
+  // This is a regression test for an issue introduced in
+  // https://github.com/ad-freiburg/qlever/pull/2252
+  using P = Result::IdTableVocabPair;
```

Comment:
These tests heavily rely on the (default) block size of the index  in the test conig. better set this manually and also document which elements you expect to be in which block.